jerry-core/rcs/rcs-allocator.cpp

```diff
@@ -0,0 +1,395 @@
+/* Copyright 2015 Samsung Electronics Co., Ltd.
+ * Copyright 2015 University of Szeged
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "rcs-allocator.h"
+#include "rcs-records.h"
+
+/**
+ * Assert that recordset state is correct.
+ */
+static void
+rcs_assert_state_is_correct (record_set_t *rec_set_p __attr_unused___) /**< recordset */
+{
+#ifndef JERRY_DISABLE_HEAVY_DEBUG
+  size_t node_size_sum = 0;
+  size_t record_size_sum = 0;
+
+  record_t *last_record_p = NULL;
+  record_t *rec_p;
+  record_t *next_rec_p;
+
+  for (rec_p = rcs_record_get_first (rec_set_p);
+       rec_p != NULL;
+       last_record_p = rec_p, rec_p = next_rec_p)
+  {
+    JERRY_ASSERT (rcs_record_get_size (rec_p) > 0);
+    record_size_sum += rcs_record_get_size (rec_p);
+
+    rcs_chunked_list_t::node_t *node_p = rec_set_p->get_node_from_pointer (rec_p);
```

Comment:
I already have a patch but I wanted to upload it as a follow up patch. Or should I merge with this?